api/renewer.go

```diff
@@ -0,0 +1,280 @@
+package api
+
+import (
+	"errors"
+	"math/rand"
+	"sync"
+	"time"
+)
+
+func init() {
+	// Seed the random generator
+	rand.Seed(time.Now().Unix())
```

Comment:
Since this is intended to be imported by other projects, should we do this? It's a bit odd because the importer may have already seeded from another source. I wonder if there is any real downside to using the predictable math/rand for this anyways.